ext/json/json.c

```diff
@@ -35,6 +36,46 @@ PHP_JSON_API zend_class_entry *php_json_exception_ce;
 
 PHP_JSON_API ZEND_DECLARE_MODULE_GLOBALS(json)
 
+json_error_info_type json_error_info_data = {
+	NULL, // Parse token
+	0, // Error ode
+	0, // Character count
+	0 // Total number of characters to parse
+};
+
+void update_json_error_info_data(php_json_parser *parser) {
+    json_error_info_data.errcode = parser->scanner.errcode;
+    json_error_info_data.character_count = parser->scanner.character_count;
+    json_error_info_data.character_max_count = parser->scanner.character_max_count;
+
+    if (json_error_info_data.token) {
+        efree(json_error_info_data.token);
+    }
+
+    char token[32];
+    snprintf(
+        token,
+        sizeof(token),
+        "%s",
+        parser->scanner.token
+    );
```

Comment:
Do we want to *print* something in the function called to *update* data? This might very well be justified, I just don't understand it for now, and so I wanted to ask.